### ISSUE TITLE
Systemd 247 crashes when exiting a VM

### ISSUE THREAD

**Issue by terencode** [systemd/systemd#17758]
**systemd version the issue has been seen with**
 > systemd 247

**Used distribution**
 > ArchLinux

**Linux kernel version used** (`uname -a`)
 > 5.9.9-95-tkg-bmq  TKG SMP PREEMPT 

**CPU architecture issue was seen on**
 > x86_64, (ryzen 3600)

**Expected behaviour you didn't see**
 > When exiting a VM, system resumes normal operations.

**Unexpected behaviour you saw**
 > When exiting a VM, systemd (from the host) crashes.

**Steps to reproduce the problem**
 > I'm using a Windows 10 VM and dynamically isolate CPUs using the script found here: https://rokups.github.io/#!pages/gaming-vm-performance.md which is only compatible with cgroups v1.

There is no such behaviour on systemd 246.6 (downgrading fixed it).

Let me know if more information is needed.

**Backtrace**

```
#0  0x00007fae2551bff0 in main_arena () from /usr/lib/libc.so.6
#1  0x00007fae2576c1f8 in base_bucket_hash (h=h@entry=0x7fae2551c010 <main_arena+1552>, p=p@entry=0x27e3) at ../systemd-stable/src/basic/hashmap.c:341
#2  0x00007fae2576ea4b in _hashmap_remove_value (h=0x7fae2551c010 <main_arena+1552>, key=key@entry=0x27e3, value=value@entry=0x55bc056c8660) at ../systemd-stable/src/basic/hashmap.c:1523
#3  0x000055bc0388fd6e in hashmap_remove_value (value=0x55bc056c8660, key=0x27e3, h=<optimized out>) at ../systemd-stable/src/basic/hashmap.h:206
#4  unit_unwatch_pid (u=0x55bc056c8660, pid=10211) at ../systemd-stable/src/core/unit.c:2818
#5  0x000055bc0385927a in manager_invoke_sigchld_event (m=<optimized out>, u=0x55bc056c8660, si=0x7ffe63acd9f0) at ../systemd-stable/src/core/manager.c:2510
#6  0x000055bc038596e7 in manager_dispatch_sigchld (source=<optimized out>, userdata=0x55bc055a8a70) at ../systemd-stable/src/core/manager.c:2589
#7  0x00007fae25805a0a in source_dispatch (s=0x55bc055aa030) at ../systemd-stable/src/libsystemd/sd-event/sd-event.c:3278
#8  0x00007fae2580937b in sd_event_dispatch (e=e@entry=0x55bc055a90e0) at ../systemd-stable/src/libsystemd/sd-event/sd-event.c:3689
#9  0x00007fae25809fd1 in sd_event_run (e=0x55bc055a90e0, timeout=18446744073709551615) at ../systemd-stable/src/libsystemd/sd-event/sd-event.c:3747
#10 0x000055bc0385de93 in manager_loop (m=0x55bc055a8a70) at ../systemd-stable/src/core/manager.c:2976
#11 0x000055bc0381ddb4 in invoke_main_loop (ret_error_message=<optimized out>, ret_switch_root_init=<optimized out>, ret_switch_root_dir=<optimized out>, ret_fds=<optimized out>, ret_shutdown_verb=<optimized out>, ret_retval=<optimized out>, ret_reexecute=<optimized out>, saved_rlimit_memlock=<optimized out>, saved_rlimit_nofile=<optimized out>, m=<optimized out>) at ../systemd-stable/src/core/main.c:1860
#12 main (argc=<optimized out>, argv=0x7ffe63acdee8) at ../systemd-stable/src/core/main.c:2861
```

**Comment by keszybz** [comment]
Please install debug symbols and provide a proper backtrace. We can't do anything without that.

**Comment by terencode** [comment]
> Please install debug symbols and provide a proper backtrace. We can't do anything without that.

Yup my bad, I edited the post.

**Comment by keszybz** [comment]
Should be fixed by e9eec8b5d2, or v247.1 when it's released later today. Please reopen if not.
